packages/mui-material/src/Button/Button.js

```diff
@@ -339,7 +340,7 @@ const Button = React.forwardRef(function Button(inProps, ref) {
   return (
     <ButtonRoot
       ownerState={ownerState}
-      className={clsx(className, contextProps.className)}
+      className={clsx(contextProps.className, classes_root, className)}
```

Comment:
Nice catch, this was broken as the classes were being spread to the `ButtonBase` component. I would propose adding this on at least one more component, to verify that it works.

To clarify why this is important. Is emotion considering the classes order for injecting the styles in the correct order? If this is the case we probably want to add an utility that will do this consistently in all components. cc @michaldudak we probably will also need to make sure the classes are in expected order in the `useSlotProps` hook too just to avoid surprises.

Reply:
> I would propose adding this on at least one more component, to verify that it works.  

Ok! Will do.

> Is emotion considering the classes order for injecting the styles in the correct order?  

Yes, precisely

Reply:
Yep, looks like we will need to update this logic. It's not great that base depends on this, but I don't have better idea for fixing the issue at this moment.

Reply:
I've created #33383 to address this issue.